src/components/account-deletions-report.tsx

```diff
@@ -0,0 +1,84 @@
+import { useContext, useRef } from "preact/hooks";
+import { ascending, descending, flatGroup } from "d3-array";
+import * as Plot from "@observablehq/plot";
+import useRegistrationData from "../hooks/use-registration-data";
+import useElementWidth from "../hooks/use-element-width";
+import { ReportFilterContext } from "../contexts/report-filter-context";
+import { formatAsPercent, yearMonthDayFormat } from "../formats";
+import PlotComponent from "./plot";
+import Table from "./table";
+import type { ProcessedResult } from "../models/daily-registrations-report-data";
+import type { TableData } from "./table";
+import { format } from "d3-format";
+
+interface ProcessedFormattedData {
+  date: Date;
+  value: number;
+}
+
+function plot({ data, width }: { data: ProcessedFormattedData[]; width?: number }): HTMLElement {
+  return Plot.plot({
+    color: {
+      type: "ordinal",
+      scheme: "Tableau10",
+    },
+    marks: [Plot.ruleY([0]), Plot.line(data, { x: "date", y: "value" })],
+    width,
+    y: {
+      domain: [0, 0.1],
+    },
+  });
+}
+
+function tabulate(results: ProcessedFormattedData[]): TableData {
+  return {
+    header: ["Date", "Percent"],
+    body: results
+      .sort(({ date: aDate }, { date: bDate }) => ascending(aDate, bDate))
+      .map(({ date, value }) => [yearMonthDayFormat(date), value]),
+  };
+}
+
+function getStartOfWeek(date: Date) {
+  const day = date.getDay();
+  const offset = date.getDate() - day;
+  const startOfWeek = new Date(date);
+  startOfWeek.setDate(offset);
+  return startOfWeek;
+}
```

Comment:
we can use a d3 date helper for this (and make sure it works in UTC too), d3.interval.floor  (so `d3.utcWeek.floor` here)

https://github.com/18F/identity-reporting/blob/d78825816f3193775a8fb8ca3cdcc9c1ff1d7c78/src/routes/report-route.tsx#L83

and if we wanted the week to start on Monday there's also [`utcMonday`](https://github.com/d3/d3-time#timeMonday) too etc

Reply:
Super helpful, thanks 👍

Reply:
> we can use a d3 date helper for this (and make sure it works in UTC too), d3.interval.floor (so `d3.utcWeek.floor` here)

Switched to `d3-time` helper in 36e1ac4.